main.js

```diff
@@ -0,0 +1,446 @@
+
+        detailValueElements.windspeed.textContent = `${data.wind.speed} m/s`;
+        detailValueElements.humidity.textContent = `${data.main.humidity}%`;
+        detailValueElements.feelslike.textContent = `${Math.round(data.main.feels_like)}°C`;
+        detailValueElements.pressure.textContent = `${data.main.pressure} hPa`;
+
+        const weatherCondition = data.weather[0].main;
+        const iconFilename = weatherIcons[weatherCondition] || 'default.png';
+        weatherIcon.src = `icons/${iconFilename}`;
+        weatherIcon.alt = weatherCondition;
+
+        const sunrise = new Date(data.sys.sunrise * 1000);
+        const sunset = new Date(data.sys.sunset * 1000);
+        const now = new Date();
+
+        if (now >= sunrise && now <= sunset) {
+            document.body.style.backgroundImage = "url('https://images.unsplash.com/photo-1503803548695-c2a7b4a5b875?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MTR8fHN1bnJpc2V8ZW58MHx8MHx8fDA%3D')";
+        } else {
+            document.body.style.backgroundImage = "url('https://media.istockphoto.com/id/162515751/photo/moon-over-mountains.webp?a=1&b=1&s=612x612&w=0&k=20&c=Xp_CfaVfyn9yJvTZtXzRBdIABemq-dztHBUZ5ZYmqtU=')";
+        }
+        document.body.style.backgroundSize = "cover";
+    } else {
+        console.error('Incomplete weather data:', data);
+        errorMessage.textContent = 'Weather data is missing or incomplete.';
+        errorMessage.style.display = 'block';
+        loadingSpinner.style.display = 'none';
+    }
+}
+
+
+searchButton.addEventListener('click', function () {
+    const city = searchInput.value;
+
+    weatherInfo.style.display = 'none';
+    errorMessage.style.display = 'none';
+
+    loadingSpinner.style.display = 'block';
+    if (city.trim() === '') {
+        errorMessage.textContent = ' Enter a city name.';
+        errorMessage.style.display = 'block';
+        loadingSpinner.style.display = 'none';
+        return;
+    }
+
+    loadingSpinner.style.display = 'block';
+    fetchData('weather', city);
+
+});
+
+function fetchData(endPoint, city) {
+    fetch(`https://api.openweathermap.org/data/2.5/${endPoint}?q=${city}&appid=${apiKey}&units=metric`)
+
+        .then(response => {
+            if (!response.ok) {
+                throw new Error('City not found');
+            }
+            return response.json();
+        })
+        .then(data => {
+            displayWeatherData(data);
+            console.log(data);
+            loadingSpinner.style.display = 'none';
+        })
+
+        .catch((error) => {
+            console.error('Error fetching weather data:', error);
+            errorMessage.textContent = 'city not found.';
+            errorMessage.style.display = 'block';
+            loadingSpinner.style.display = 'none';
+        });
+}
+
+function fetchHourlyForecast(city) {
+    fetch(`https://api.openweathermap.org/data/2.5/forecast?q=${city}&appid=${apiKey}&units=metric`)
```

Comment:
https://api.openweathermap.org/data/2.5 this value can also be store in constant